src/test/java/org/apache/sysml/test/integration/functions/paramserv/ParamservFuncTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysml.test.integration.functions.paramserv;
+
+import org.apache.sysml.api.DMLException;
+import org.apache.sysml.test.integration.AutomatedTestBase;
+import org.apache.sysml.test.integration.TestConfiguration;
+import org.junit.Test;
+
+public class ParamservFuncTest extends AutomatedTestBase {
```

Comment:
Related to the IPA extension, it would be good to add a test where paramserv is not used immediately under a DAG root node. For example, you could so something like `print(length(paramserv()))`.

Reply:
In fact, I have tried ```print(length(paramserv()))````but it won't add the paramserv dag into the tree. Because I found that the builtin length function could directly get the length of the paramserv output without executing the paramserv function. Except that I try to print the element of the result list where it will add the paramserv dag.

Reply:
good point - in these situations I usually add a if/else with some data-dependent conditions (e.g., sum(A)>7) and the assignment of lists with different lengths into if and else which causes that the length cannot be evaluated during runtime. You might also need to disable recompilation to actually execute the paramserv function.